fission-core/library/Fission/App/Types.hs

```diff
@@ -0,0 +1,64 @@
+module Fission.App.Types
+  ( module Fission.App.Name
+  , App(..)
+  ) where
+
+import           Data.Swagger     hiding (URL, url)
+
+import           Fission.Prelude
+
+import           Fission.App.Name
+import           Fission.URL
+
+
+data App = App
+  { urls       :: [URL]
+  , insertedAt :: UTCTime
+  , modifiedAt :: UTCTime
+  }
+  -- What about deriving Json of normal data?
+  -- deriving (Eq, Generic, Show, ToJSON)
+  deriving (Eq, Show)
```

Comment:
We may serialize the app in a bunch of different ways, because we're decoupling the inner representation from the serialized view. In Rails or Phoenix terms, these are the "model" and "view" layers.

I'd maybe put this in `Fission.Web.API.App.Detail.Payload.Types` or similar, since it's related to the specific endpoint, not all places we need `App`s.

Reply:
Oh, also something that you would have no way of knowing from just looking at the code: we're going to be reworking the app model in the near future (maybe even you'll get to it!) to have one canonical Fission-subdomain, and several "follow"ers (there's some IPFS magic in here that we get into later). If you can grab the first URL. Something like one of these two:

```haskell
data Payload = Payload
  { url :: URL
  , insertedAt :: UTCTime
  , modifiedAt :: UTCTime
  }
```

```haskell
data Payload = Payload
  { subdomain :: Subdomain
  , host :: Host
  , insertedAt :: UTCTime
  , modifiedAt :: UTCTime
  }
```